packages/xsnap/src/xsnap.js

```diff
@@ -200,7 +206,30 @@ export function xsnap(options) {
           )}`,
         );
       } else if (message[0] === QUERY) {
-        await messagesToXsnap.next(await handleCommand(message.subarray(1)));
+        // This nested await is safe because "not-my-problem".
+        //
+        // This nested await introduces no unsafety beyond that introduced
+        // by the next await, since it is earlier within the same control
+        // flow branch, and is top level within that branch.
+        // eslint-disable-next-line @jessie.js/no-nested-await
+        const array = await handleCommand(message.subarray(1));
+        // This nested await is ALMOST safe because "terminal-control-flow".
+        // This one appears nested in an unbalanced conditional, so we need
+        // to consider three cases:
+        //   * This branch is taken and the awaited expression throws.
+        //     In that case, the function exits would executing and further
+        //     code.
+        //   * This branch is taken and does not throw. In that case,
+        //     it causes a turn boundary before the next iteration.
+        //   * Another branch of the conditional is taken, skipping
+        //     this statement and proceeding syncronously to the next
+        //     iteration. The await at the top of the loop ensures that the
+        //     remainder of the loop only happens after a turn boundary.
+        //     The only remaining unsafety is that the awaited expression of
+        //     that initial await may therefore be reached synchronously or
+        //     asynchronously.
```

Comment:
The expression awaited at the top of the cthuloop, `messagesFromXsnap.next(undefined)`, just manipulates an entangled pair of async promise queues, putting undefined on one and getting the next promise from the other. I’m not sure whether this affects your analysis.